src/Microsoft.IdentityModel.Tokens/Encryption/EcdhKeyExchangeProvider.cs

```diff
@@ -0,0 +1,275 @@
+                    KeyDataLen = 128;
+                else if (SecurityAlgorithms.Aes192CbcHmacSha384.Equals(enc, StringComparison.InvariantCulture))
+                    KeyDataLen = 192;
+                else if (SecurityAlgorithms.Aes256CbcHmacSha512.Equals(enc, StringComparison.InvariantCulture))
+                    KeyDataLen = 256;
+            }
+            else
+            {
+                _algorithmId = alg;
+
+                if (SecurityAlgorithms.EcdhEsA128kw.Equals(alg, StringComparison.InvariantCulture))
+                    KeyDataLen = 128;
+                else if (SecurityAlgorithms.EcdhEsA192kw.Equals(alg, StringComparison.InvariantCulture))
+                    KeyDataLen = 192;
+                else if (SecurityAlgorithms.EcdhEsA256kw.Equals(alg, StringComparison.InvariantCulture))
+                    KeyDataLen = 256;
+            }
+        }
+
+        private static void ValidateAlgAndEnc(string alg, string enc)
+        {
+            if (string.IsNullOrEmpty(alg))
+                throw LogHelper.LogArgumentNullException(alg);
+            if (string.IsNullOrEmpty(enc))
+                throw LogHelper.LogArgumentNullException(enc);
+
+            if (!SupportedAlgorithms.EcdsaWrapAlgorithms.Contains(alg) && !SecurityAlgorithms.EcdhEs.Equals(alg, StringComparison.InvariantCulture))
+                throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10652, LogHelper.MarkAsNonPII(alg))));
+
+            if (!SupportedAlgorithms.SymmetricEncryptionAlgorithms.Contains(enc))
+                throw LogHelper.LogExceptionMessage(new NotSupportedException(LogHelper.FormatInvariant(LogMessages.IDX10715, LogHelper.MarkAsNonPII(enc))));
+        }
+
+        private void ValidateCurves(string privateKeyArgName, string publicKeyArgName)
+        {
+            if (_ecParamsPrivate.Curve.Equals(_ecParamsPublic.Curve))
+            {
+                throw LogHelper.LogArgumentException<ArgumentException>(
+                    privateKeyArgName,
+                    LogHelper.FormatInvariant(
+                        LogMessages.IDX11000,
+                        LogHelper.MarkAsNonPII(privateKeyArgName),
+                        LogHelper.MarkAsNonPII(_ecParamsPrivate.Curve.ToString()),
+                        LogHelper.MarkAsNonPII(publicKeyArgName),
+                        LogHelper.MarkAsNonPII(_ecParamsPublic.Curve.ToString()))
+                    );
+            }
+        }
+
+        private void Initialize()
+        {
+            _ecdhPublic = ECDiffieHellman.Create(_ecParamsPublic);
+            _ecdhPrivate = ECDiffieHellman.Create(_ecParamsPrivate);
+        }
+
+        private static byte[] Concat(params byte[][] arrays)
+        {
+            int length = 0;
+            foreach (byte[] arr in arrays)
+                length += arr.Length;
+
+            byte[] output = new byte[length];
+            int x = 0;
+            foreach (byte[] arr in arrays)
```

Comment:
Can we use ArrayCopy as we do here: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/039d09b7bc4c1671e1c4093d5a62d307c2f96935/src/Microsoft.IdentityModel.Tokens/Encryption/SymmetricKeyWrapProvider.cs#L280

Reply:
Changed in a52ecf360f5d0463192c760898765e2a8542ee68